addons/utm/models/utm_campaign.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import fields, models, api, SUPERUSER_ID
+
+
+class UtmCampaign(models.Model):
+    _name = 'utm.campaign'
+    _description = 'UTM Campaign'
+
+    name = fields.Char(string='Campaign Identifier', required=True, compute='_compute_name', store=True, readonly=False)
+    full_name = fields.Char(string='Campaign Name', required=True, translate=True)
```

Comment:
Well actually it is probably better to keep name and identifier actually, you are right on this. That way people set the name (that is translatable), and it can be used to generate the identifier.